Use presets to simplify configuration and build commands

ChrisThrasher opened the pull request:
CMake presets are the perfect tool to capture the particular flags required to build the `develop2` branch. This repo ought to set an example for how presets can be used to avoid typing out long configuration commands.

ChrisThrasher committed:
Use presets to simplify configuration and build commands

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/cmake-conan?pullRequest=505) <br/>All committers have signed the CLA.

ChrisThrasher committed:
Remove unnecessary platform-specific command

The `-B build` option means that a build/ directory will automatically
get created. This has the benefit of removing code and ensuring this
example is more portable to OSes that don't define `mkdir`.

memsharded commented on README.md:
````diff
@@ -25,10 +25,8 @@ git clone https://github.com/conan-io/cmake-conan.git -b develop2
 This repository contains a `CMakeLists.txt` with an example project that depends on `fmt`. 
 
 ```bash
-cd cmake-conan
````
The ``cd cmake-conan`` is still necessary, we are not yet inside the cloned project

memsharded reviewed (commented):
thanks for your contribution!

memsharded commented on README.md:
```diff
@@ -26,9 +26,8 @@ This repository contains a `CMakeLists.txt` with an example project that depends
-mkdir build
-cmake -B build -S . -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=conan_provider.cmake -DCMAKE_BUILD_TYPE=Release
-cmake --build build --config Release
+cmake --preset default
```
Let's leave both approaches.
I think it is good that we provide both, with and without presets, for users that don't want to use presets. Could you please leave it?

memsharded commented:
Hi @ChrisThrasher 
Did you accidentally close the PR? Sorry for the delay, it didn't mean that it was not an interesting contribution, but mostly that we didn't have the time to manage it.

I would still like @jcar87 feedback if he feels it would be good to add these presets.

ChrisThrasher commented:
I closed it because I lost motivation to complete it and didn't want lingering open PRs around.

memsharded commented:
Ok, no prob, we might take the idea of adding these presets and add it ourselves. Thanks anyway!
